hs_core/views/resource_folder_hierarchy.py

```diff
@@ -217,6 +220,11 @@ def data_store_structure(request):
                      'aggregations': aggregations,
                      'can_be_public': resource.can_be_public_or_discoverable}
 
+    if found_unreferenced_files:
+        from hs_core.management.utils import ingest_irods_files
+        ingest_irods_files(resource, None)
+        return data_store_structure(request)
```

Comment:
This recursion is convenient but my only concern is that it might be slow for large folder hierarchies.
Because the `continue` L149 will keep iterating even after `found_unreferenced_files` is True. But then after iterating all of the files we get down to L223 and we will ingest the missing files and then iterate ALL files in the resource all over again.

Reply:
Do you have a preference on how to approach this? This is straightforward and is only a one time cost, when an unreferenced file is found. The code will only execute again if a new unreferenced file is found.

We could kick this off in an async task if you're concerned it will take a long time to run. My sense on this is that this is better than what we have is a minor on time performance cost for the benefit, though I'm open to working through any concerns.

Reply:
Sounds reasonable!